attendance/API_HR/c-hr.php

```diff
@@ -6086,6 +6086,162 @@ public static function getEmployeesHistoryStats(){
         return $return;
     }
 
+    public static function validateSecretKey($secret_key){
+        $return = false;
+        $q = " SELECT * from secret_tokens WHERE secret_key = '$secret_key' ";
+        $runQuery = self::DBrunQuery($q);
+        $row = self::DBfetchRows($runQuery);
+        if( sizeof($row) > 0 ){
+            $return = true;
+        }
+        return $return;
+    }
+
+    public static function generateSecretKey($length = 60){
+        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
```

Comment:
this will leads the hacker to generate combinations using these characters and he can easily crack our system